lib/fork.js

```diff
@@ -59,7 +61,54 @@ class SharedWorkerChannel extends Emittery {
 
 let forkCounter = 0;
 
+const createWorker = (options, execArgv) => {
+	let worker;
+	let port;
+	let _send;
+	let _exit;
+	if (options.workerThreads) {
+		const {port1, port2} = new MessageChannel();
+		worker = new Worker(WORKER_PATH, {
+			argv: options.workerArgv,
+			env: {NODE_ENV: 'test', ...process.env, ...options.environmentVariables, AVA_PATH},
+			execArgv,
+			workerData: {
+				port: port1,
+				cwd: options.projectDir
```

Comment:
Could add the options here?

Though we may want to clear this object before loading user code, to discourage these internals being used.

Reply:
I see what you mean. Yes, we can pass the options here, but I see a few drawbacks:

1. This current approach it's good if we want to use a worker thread pool implementation in the future (worker threads are not  created each time but shared. Hence we need a way to pass options to an already initialized worker)
2. I am afraid this change would force us to create 2 separate files & APIs for the worker file imported in the tests, one for the subprocess and one for the worker thread.

Reply:
Threads should be cheap enough to recreate. Reuse leads to modifications from a previous test file surviving into the next run which isn't ideal.